### DISCUSSION TITLE
Technical Conferences to present Fairlearn

### DISCUSSION THREAD

**Discussion by LeJit** [fairlearn/fairlearn#674] [General]
This year, we want to promote FairLearn more through talks and tutorials at developer-oriented conferences. The goal of this discussion thread is to crowd-source a list of conferences (and associated deadlines for CfPs) that we want to target. Some upcoming conferences we should consider:

- [ODSC East](https://odsc.com/boston/call-for-speakers/)
    - Talk Deadline: February 7, 2021
- SciPy 2021
    - Talk Deadline: February 9, 2021
    - Tutorial Deadline: February 9, 2021
- [PyCon 2021](https://us.pycon.org/2021/)
    - Talk Deadline: February 12, 2021
    - Tutorial Deadline: February 12, 2021

**Comment by LeJit** [comment]
# PyCon 2021 Talk Proposal

## Title 
Assessing and Mitigating Unfairness in AI Systems

## Who and Why?
This talk is intended primarily for data scientists and machine learning researchers who want to incorporate unfairness mitigation best practices and algorithms into their workflow. We expect audience to have a basic understanding of supervised ML algorithms and data science processes.

After watching this talk, I hope audience members have a better understanding of fairness-related harms and how to approach the sociotechnical challenges of fairness in applied context. I provide a qualitative framework for assessing harms that audience members can use to assess their own projects for unfairness, and showcase various techniques in the Fairlearn library that audience members can install and use in their workflow.

## Description
While fairness has become an increasingly popular topic in machine learning and data science, many data scientist struggle with how to incorporate fairness assessments and unfairness techniques into their work. We will cover formal frameworks for assessing ML systems for fairness-related harms and how to apply different algorithmic techniques for mitigating unfairness in trained models.

## Outline
- Introduction (1 Minute)
- Overview of Fairness and Responsible AI (4 Minutes)
    - Examples of Fairness Harms
        - ProPublica/COMPASS
        - Weapons of Math Destruction
    - Fairness should focus on impact on people, not bias in models
- Fives types of Fairness Harms (5 Minutes)
    - Resource Allocation
    - Quality-of-Service 
    - Stereotyping
    - Denigration
    - Over/Under-representation
- Unfairness Mitigation Techniques (6 minutes)
    - Post-Processing Mitigation Techniques (3 minutes)
        - Overview of Post Processing algorithm
        - Pros/Cons of this approach
    - In-Processing Mitigation Techniques (3 minutes)
        - Overview of Reduction algorithm
        - Pros/Cons of this approach
- Case Study: Speech-to-Text Transcription (12 minutes)
    - Overview of Speech-To-Text Transcription (Alexa, Cortana, Siri)
    - Identify the affected groups
        - [Studies show smart assistants display disparities in performance by race](https://www.nytimes.com/2020/03/23/technology/speech-recognition-bias-apple-amazon-google.html)
    - Identify the fairness-related harms
        - Quality-of-service deterioration
    - Define the fairness metric 
        - Disaggregated Word Error Rates (WER) by racial identity
    - Applying the In-Processing mitigation technique
- Conclusion (2 minutes)
    - Summary 
    - Further Resources

**Reply by MiroDudik** [reply]
@LeJit and I have chatted about this and have come up with this revised and slightly more streamlined outline (see below). Two key simplifications are: (1) we just give examples of representational harms (without subdividing), and (2) we don't talk about postprocessing (since we don't have a matching example). Thoughts?
 
Revised outline:
1. [4 slides] Open with three examples (that will be later used to motivate the harms): allocational (hiring/lending or ProPublica/COMPAS), speech example, and some representational
2. [1 slide] Many ways to define fairness; we have found it most useful to center it around "harms to people" rather than motivations / intentions.
3. [1 slide] Three categories of harms -- link back to examples; note that these are not mutually exclusive and can co-occur and exacerbate each other.
4. [1-2 slides] How do we assess and mitigate harms? There are many points in the AI life cycle [use one that is not too long... 5 stages?] where one could intervene.
   * Describe the stages
   * Provide some links to different kinds of tools for these stages (?), but don't get into the details
5. [0.5 slide] In the rest of the talk we focus on model design (assessment & algo. mitigation) plus assessment post-deployment (?)
6. [1 slide] Maybe: show which tools are applicable to model design & assessment post-deployment split by the type harm
   * The point to make is that there are quite a few tools for allocation, fewer for QoS, fewer for representational
7. [0.5 slide] The rest of the talk on QoS harms.
8. [1 slide] Zoom-in to the Word Error Rate example
9. [3 slides] ASSESSMENT for QoS
10. [3 slides] MITIGATION for QoS
11. [1 slide] Conclusion/Summary
12. [additional slides for further resources]

TOTAL: 17-18 slides

**Reply by LeJit** [reply]
# Title
Assessing and mitigating unfairness in AI systems

# Who and Why? 
This talk is intended primarily for data scientists and machine learning practitioners who want to incorporate fairness assessment and unfairness mitigation best practices into their workflow. Audience members should have a basic understanding of supervised ML algorithms and data science processes.

After the talk, audience members will have a better understanding of different types of fairness-related harms and strategies for assessing and mitigating them, including using the Fairlearn library to do this.

# Description
Although the fairness of AI systems has become an increasingly popular topic in both data science and machine learning, many practitioners struggle with how to incorporate fairness assessment and unfairness mitigation into their workflows. Drawing from real-world examples, we present a framework for assessing and mitigating fairness-related harms in AI systems. Through a concrete use case, we show how to apply this framework in settings where AI systems, such as face-detection or speech-to-text systems, fail to provide a similar quality of service to different groups of people. We also demonstrate how to use the Fairlearn library to assess fairness and mitigate unfairness in these settings.

# Outline

- 	Introduction (1 minute)
- 	Overview of fairness and responsible AI (6 minutes)
    - 	Examples of fairness-related harms (2 minutes)
        - 	[Apple Credit Card story](https://www.bbc.com/news/business-50365609)
        - 	[Speech-to-text transcription](https://www.nytimes.com/2020/03/23/technology/speech-recognition-bias-apple-amazon-google.html)
        - 	Google Photos story
- 	Many definitions of “fairness”, but we found it best to center discussion around “harms to people” (4 Minutes)
   - 	Taxonomy of fairness-related harms
      - 	Allocation of resources and opportunities
      - 	Quality of service
      - 	Representation
- 	A single system (or model) may cause multiple types of harms
- 	AI Lifecycle and fairness interventions (5 minutes)
    - 	AI lifecycle (3 minutes)
        - 	Project definition and scope
        - 	Data collection and pre-processing
        - 	Feature engineering
        - 	Model training
        - 	Model evaluation
        - 	Model deployment
    - 	Overview of different fairness tools/libraries/services (1.5 minutes)
       - 	Responsible AI tools/libraries (1 minute)
       - 	Responsible AI companies (0.5 minutes)
      - Most tools focus on allocation harms. There are fewer tools for quality-of-service harms, and even fewer for representational harms. (0.5 minutes)
      - Rest of talk will focus on model training and evaluation using Fairlearn.
- Case study: Speech-to-text transcription (12 minutes)
    - Quick recap of speech-to-text example (introduced in “Examples of Fairness-related Harms”)
    - Overview of *word error rate* metric (1 minute)
    - Disaggregated metrics (2 minutes)
    - Disparities in quality of service across groups (3 minutes)
        - Max differences between group
        - Maximal ratio between group
    - Unfairness mitigation for quality of service (5 minutes)
         - Reductions approach
             - Overview of reductions approach
             - Advantages and limitations of this approach
- Conclusion (1 minute)
   - Further resources

**Comment by trivenigandhi** [comment]
## SciPy 2021 Tutorial Proposal
# ML Fairness in the Wild : From Social Context to Practice using Microsoft's Fairlearn 

## Short Description
ML Fairness is an area of growing research and practice, one that all AI developers and practitioners should be familiar with.
In this tutorial, we cover the basics of social bias in datasets and how models are built using an example from the healthcare
industry. The hands on tutorial will make use of scikit learn and the new Fairlearn package so that students can apply key
concepts right away. The tutorial requires intermediate python skills and familiarity with Jupyter notebooks.

## Long Description
ML Fairness is an area of growing research and practice, one that all AI developers and practitioners should be familiar with.
In this tutorial, we cover the basics of social bias in datasets and how models are built using an example from the healthcare
industry. The hands on tutorial will make use of scikit learn and the new Fairlearn package so that students can apply key
concepts right away. The tutorial requires intermediate python skills and familiarity with Jupyter notebooks.

Below is an outline of the entire tutorial, including concepts and hands on aspects that will be covered.

#### ML Fairness in the Responsible AI Pipeline (30 minutes)

Before diving into the mathematical and code-based approaches to Fair ML it is critical that all practitioners understand
the social context from which we understand concepts of fairness. We will present an overview of critical concepts in
Responsible AI that serves as the foundation of ML Fairness. 

##### Data is Never Neutral 

CONCEPT : Data is a product of the historic and social context in which is it measured, and how it is measured. This section will go
over some well-known areas of biased data and how they reflect in our pipelines.

##### 5 Types of Harms 

CONCEPT : Unfair AI systems produce 5 types of harms covered here with real examples. Resource Allocation, Quality-of-Service,
Stereotyping, Denigration, Over/Under-representation

##### Key Terms in ML Fairness and Use Care

CONCEPT: Sensitive vs. Advantaged groups, positive vs negative outcomes

CONCEPT : We will discuss the basic premise of the use case - bias in medical algorithms - and the various consequences of 
unfair AI on real people. 


#### Uncovering and Documenting Data Bias: 1 hr 

CONCEPT : Types of social biases in data, such as proxy variables, tainted features and disparate impact.

HANDS ON : Look for biases in the data using the example notebook and discuss. 

CONCEPT : Datasheets for Datasets

HANDS ON : Build a datasheet for the dataset at hand.

#### Break: 10 min 

#### Detecting Model Bias (1 Hr) 

CONCEPT : Types of ML Fairness and how to understand which is right for the use case. What are the various tradeoffs and 
impacts of those tradeoffs?

CONCEPT : Introduce the Fairlearn Package

HANDS ON : Build a predictive model of choice, and using the Fairlearn package assess how fairly the model performs for
sensitive groups. 

#### Break: 10 min 

#### Mitigate Unfair Models (45 minutes) 

CONCEPT : Two ways to mitgate model bias - what that means mathematically and how it influence impacts socially.

HANDS ON : Mitigating Model by optimizing for different fairness goals using Fairlearn.

CONCEPT : Model Cards 

HANDS ON : Build a model card for your mitigated model. 

#### Discussion and Conclusion (15 minutes)

Re-affirm that ML Fairness does not happen in a vacuum and that is a part of a much larger conversation of responsible 
and ethical tech. Provide resources for further reading and invite folks to join the Fairlearn community.

**Reply by riedgar-ms** [reply]
A few quick thoughts:

- 'Data is never neutral' -> Might want to highlight the "data you _didn't_ gather" too
- 'Types of harm' -> Emphasise that these are not mutually exclusive.
- 'Proxy variables' -> Highlight that these are why "Fairness through unawareness" doesn't work

**Reply by MiroDudik** [reply]
I like the overall narrative flow.

The biggest questions (not clear if these need to be decided before the proposal submission):
* Do we want to commit to classification? [positive / negative outcomes]
   * The underlying problem is risk assessment, which is actually a regression problem. My sense is that the eventual decisions are based on the "risk level" which might be either numeric, or, more likely, some thresholded version, e.g., "low / medium / high". 
* Do we want to call out both allocational and quality-of-service harms in this setting?

A few lower level issues that I think need to be incorporated into the proposal:
* Hopefully, we'll move away from "Microsoft's Fairlearn" very soon (I have just been swamped this week, so I haven't yet completed the governance PR).
* We should tweak the writing to follow our [style guide](https://fairlearn.github.io/v0.5.0/contributor_guide/how_to_talk_about_fairness.html). Most notably, we should be more careful how we use the word "bias". We should decide how to coordinate this... it might be easier for me to make an editing iteration and we can discuss the specific suggested changes?

**Reply by trivenigandhi** [reply]
@MiroDudik @LeJit @romanlutz @hildeweerts @mmadaio See here for the google doc with the proposal text. Please request permission to edit, and share with others I may have missed!

https://docs.google.com/document/d/13Al6ot1BMJrvuiG1T8qVq7IkX9l8KCJUfdqqcPOzFbA/edit?usp=sharing

**Comment by LeJit** [comment]
# PyData Global

### Kind
Workshop

### Title
*Assessing and Mitigating Unfairness in AI Systems*

### Brief Summary

This workshop aims to help data science practitioners navigate the sociotechnical challenges of AI fairness. In the first half of the workshop, we walk participants through a Jupyter notebook showing how Fairlearn can be used to assess and mitigate unfairness in ML models. In the second half, a panel of speakers will discuss best practices for improving fairness of real-world AI systems.

### Brief Outline 

- **Tutorial**: Hands-on practice with Fairlearn (1 hour)
    - Overview of fairness in AI systems (10 minutes)
        - Types of fairness-related harms (6 minutes)
        - Who is affected? (4 minutes)
    - Intro to healthcare scenario (10 minutes)
        - Exploratory data analysis
        - Discussion of target variable
    - Assessing model fairness using MetricFrame (20 minutes)
        - Introduction to fairness metrics
        - Train a logistic regression model
        - Assess the logistic regression model using MetricFrame
    - Mitigating fairness-related harms in ML models (15 minutes)
         - Post-processing mitigation using ThresholdOptimizer
         - Reductions approach using ExponentiatedGradient
    - Tutorial conclusion (5 minutes)


- **Tutorial Q&A** (15 minutes)
    - Mini-panel to address audience questions from the tutorial.


- **Panel Discussion**: Responsible AI in the real-world (45 minutes)
    - Moderator: Miro Dudik
    - Tentative Panelist:
        - Hanna Wallach (Senior principal research manager at Microsoft Research)
        - Kristen Laird (Project manager at Microsoft)
        - Michael Madaio (Postdoctoral researcher at Microsoft Research)

### Description

Fairness in AI systems is an interdisciplinary field of research and practice that aims to understand the negative impacts of AI, with an emphasis on improving and supporting historically marginalized and underserved communities. In this workshop, we first walk participants through an hour-long tutorial on assessing and mitigating fairness-related harms in the context of an U.S. healthcare scenario. Participants will learn how to use the [Fairlearn](https://fairlearn.org/) library to assess machine learning models for performance disparities across different racial groups. 

In the second part of this workshop, we invite researchers and industry practitioners to speak on a panel about sociotechnical challenges data scientists face when applying fairness methodology to their work. The panel will be moderated by Miro Dudik, Senior Principal Researcher at Microsoft Research. The panelists will first introduce themselves, and then answer some moderator-provided questions about documenting responsible AI practices, discussing fairness within your team and organization, and incorporating fairness in the design and evaluation of AI systems.

**Reply by adrinjalali** [reply]
Love the proposal, but it'd be even better if the panel included people also from organizations other than Microsoft.

**Reply by MiroDudik** [reply]
100%. we wanted to put these down as tentatively confirmed, and a starting point that we were able to confirm on a quick turn-around, with the understanding that we should try to change the composition, so it's not majority microsoft (or at least it is more balanced). re. speakers, i was hoping to hit the following axes: (1) AI deployment, (2) research into data science practices, (3) governance.
